### DISCUSSION TITLE
Validating land lines

### DISCUSSION THREAD

**Discussion by Nedmon12** [HuluWZ/phone-formatter-eth#8] [Q&A] [answered]
does the validate function account for land lines? eg  011 5 5432

**Comment by HuluWZ** [comment, marked as answer]
Thank you for your question! Currently, the `validate` function in **phone-formatter-eth** focuses on validating mobile phone numbers in Ethiopian formats. Specifically, it checks for the validity of numbers formatted as `+251XXXXXXXXX` or `09XXXXXXXX`, which covers mobile phones.

### Handling Landline Numbers

You’re right; our existing implementation does not account for Ethiopian landline numbers, such as `011 5 5432`. We are planning to extend the package to support landline number validation and formatting. This new feature will include:

- **Landline Number Format:** Validation and formatting for landline numbers typically formatted as `011X XXXXX` or `011-5-5432`.
- **Format Conversion:** Converting between international, local, and E.164 formats for landline numbers.
- **Normalization:** Handling landline numbers with or without country codes and removing unnecessary characters.

### Invitation for Collaboration

We would love to collaborate with the community on this enhancement! If you’re interested in contributing to the development of landline number support, here’s how you can help:

1. **Feature Development:** Implement the landline number validation and formatting features. You can follow the planned feature specifications or propose additional improvements.
2. **Testing:** Ensure that the new features work correctly with various landline formats and edge cases.
3. **Pull Request:** Once you’ve developed and tested the features, please submit a pull request. We’ll review it, provide feedback, and merge it into the main codebase.

### Example Feature Implementation

Here’s a basic outline of what the implementation might look like:

```javascript
// Function to validate landline numbers
const validateLandline = (number) => {
    // Example implementation for validating landline numbers
    const landlinePattern = /^(011\d{1} \d{5}|\011-\d-\d{4})$/;
    return landlinePattern.test(number);
};

// Function to format landline numbers
const formatLandline = (number, formatType) => {
    // Example implementation for formatting landline numbers
    if (formatType === 'local') {
        return number.replace(/(\d{3}) (\d{5})/, '011 $1 $2');
    }
    // Additional formats can be implemented as needed
};

// Example usage
const landlineNumber = '011 5 5432';
if (validateLandline(landlineNumber)) {
    console.log('Landline number is valid');
} else {
    console.log('Landline number is invalid');
}
```

Getting Started

To start contributing, you can:

    Fork the Repository: Create a fork of the phone-formatter-eth repository on GitHub.
    Clone the Repository: Clone your fork to your local machine.
    Create a Branch: Create a new branch for your feature development.
    Implement and Test: Develop the feature and write tests to ensure its correctness.
    Submit a Pull Request: Push your changes to your fork and submit a pull request to the main repository.

We’re excited to work together to enhance the phone-formatter-eth package! Your contributions will help us provide a more comprehensive tool for managing Ethiopian phone numbers.

Feel free to reach out if you have any questions or need further guidance on the implementation. We look forward to collaborating with you!
